src/Matrix.jl

```diff
@@ -48,13 +48,34 @@ function _check_bases(a, b)
   return nothing
 end
 
-function (s::MatSpace{T})(a::M) where {T, M <: MatrixElem{T}}
+# create a zero matrix
+function (s::MatSpace{T})() where {T <: NCRingElement}
+  return zero_matrix(base_ring(s), nrows(s), ncols(s))::eltype(s)
+end
+
+function (s::MatSpace{T})(a::MatrixElem{T}) where {T <: NCRingElement}
   _check_dim(nrows(s), ncols(s), a)
   _check_bases(s, a)
   a isa eltype(s) && return a
-  b = eltype(s)(a)
-  b.base_ring = base_ring(s)
-  return b
+  M = s()  # zero matrix
+  R = base_ring(s)
+  for i = 1:nrows(s)
+    for j = 1:ncols(s)
+      M[i, j] = R(a[i, j])
```

Comment:
This is somewhat inefficient if `base_ring(a) == base_ring(s)` (as we perform a redundant parent check for every element). But we can always improve that in a future update if we deem it necessary.

Reply:
From my POV it would be fine to merge this as-is and then tweak it later, whenever you (or someone else) has time resp. need for it. Your choice.

Reply:
done in [`09933d8` (#1839)](https://github.com/Nemocas/AbstractAlgebra.jl/pull/1839/commits/09933d899253a464dd95ef6488b43f5f429692b1). please merge if you are happy with it